cmake/FindCharm.cmake

```diff
@@ -79,18 +109,13 @@ find_program(CHARM_COMPILER
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(
   Charm
-  FOUND_VAR CHARM_FOUND
-  REQUIRED_VARS CHARM_COMPILER CHARM_INCLUDE_DIRS CHARM_LIBRARIES
```

Comment:
Why remove the required vars?

Reply:
My understanding is that the `REQUIRED_VARS` should be those that are searched by cmake on the system, and can alternatively be specified directly on the CLI. If it makes no sense to specify it on the CLI, it shouldn't be in this list. Since the vars here should all be consistent with each other, i.e. point into the same charm_root, none should be specified on the CLI. Maybe we should actually replace `CHARM_COMPILER` with `CHARM_ROOT` as well.
Docs: https://cmake.org/cmake/help/latest/module/FindPackageHandleStandardArgs.html

Reply:
That specified use seems reasonable, and what is supported by the docs, but what I'm asking about is the practical question of what is the cmake behavior if, say, `CHARM_COMPILER` is found but `CHARM_LIBRARIES` is not (e.g. if the charm build got mangled somehow)?
My understanding from the previous code was that it would error during the configuration. Does it still do so without the extra entries in `REQUIRED_VARS`? And if we wanted that behavior, is the right thing to do to add extra manual checks and errors?

Reply:
It should be possible to override INCLUDE and LIBRARIES separately.  The headers might be installed somewhere like `/usr/include/charm-6.10`.  I used to do something like this.

Reply:
Ok. I added the two back to `REQUIRED_VARS`.